src/aead/nonce.rs

```diff
@@ -99,7 +102,7 @@ where
 
     fn new(Nonce(nonce): Nonce, initial_counter: u32) -> Self {
         let mut r = Self {
-            block: Block::zero(),
+            block: [0u8; BLOCK_LEN],
         };
         let block = unsafe { &mut r.block };
         block.overwrite_part_at(U32::NONCE_BYTE_INDEX, nonce.as_ref());
```

Comment:
I understand what you are trying to do and this is a tricky module w.r.t. this effort.

This counter code is weird in large part because of my attempt to use `Block`. Once we get rid of `Block` it doesn't need to be as weird any more. Check out https://github.com/briansmith/ring/pull/1028 to see what I mean.

Reply:
Keep crypto weird ~.~ (jk)

Will check out #1028

Reply:
> Once we get rid of Block it doesn't need to be as weird any more. Check out #1028 to see what I mean.

Will definitely check out #1028. I thought I was reproducing the previous behaviour, but can see your point about weirdness, too.